src/SymfonyCache/PurgeTagsListener.php

```diff
@@ -119,8 +119,13 @@ public function handlePurgeTags(CacheEvent $event)
 
         $tags = [];
 
-        foreach ($request->headers->get($this->tagsHeader, '', false) as $v) {
-            foreach (explode(',', $v) as $tag) {
+        // Compatibility with Symfony 3+
+        $allHeaders = $request->headers->all();
```

Comment:
i would prefer to have a reflection check if `all` takes a parameter and if it does, you can `$request->headers->all($this->tagsHeader);` and otherwise keep the old way with `get`. and add a note that when we drop support for symfony < 4.4 we can drop the check.

i prefer that over this code that is harder to read and won't get cleaned up.